### PR TITLE
Support for json path

### PR DESCRIPTION
**Pull request by caskolkm**

### CODE REVIEW

File: src/main/kotlin/nl/avisi/kotlinwebtest/expressions/Expressions.kt [hunk 1]
```diff
@@ -15,43 +15,8 @@ operator fun Expression.plus(other: Expression): CompositeExpression =
         CompositeExpression(this, other)
 
 
-class ConstantExpression(val value: String) : Expression
-
-class PropertyExpression(val name: String) : Expression {
```
Comment (caskolkm) [inline src/main/kotlin/nl/avisi/kotlinwebtest/expressions/Expressions.kt]: Unused

File: src/test/kotlin/nl/avisi/kotlinwebtest/RestDslTest.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Rule
+import org.junit.Test
+
+class RestDslTest : WebTest() {
+
+    companion object {
+        private val RANDOM_PORT = 0
+    }
+
+    var wiremock = WireMockRule(options().port(RANDOM_PORT))
```
Comment (reinkrul) [inline src/test/kotlin/nl/avisi/kotlinwebtest/RestDslTest.kt]: can't this be a 'val'?

File: src/main/kotlin/nl/avisi/kotlinwebtest/soap/Validation.kt [hunk 3]
```diff
@@ -110,14 +110,14 @@ class SoapTestStep(testCase: TestCase) : TestStep<SoapStepRequest, SoapStepRespo
             endpoint ?: configuration.defaults.endpoint
 }
 
-class XPathValidator(val xpath: String, var value: Expression? = null) : Validator<SoapStepRequest, SoapStepResponse> {
+class XPathValidator(val xpath: String, val value: Expression) : Validator<SoapStepRequest, SoapStepResponse> {
```
Comment (reinkrul) [inline src/main/kotlin/nl/avisi/kotlinwebtest/soap/Validation.kt]: We might need to revert this to be able to write a better DSL (which unfortunately currently requires mutable state at many points). I'm willing to see how this turns out though.

**Review by reinkrul — APPROVED** [review]

File: src/test/kotlin/nl/avisi/kotlinwebtest/RestDslTest.kt (hunk 2 above)
Comment (caskolkm) [reply src/test/kotlin/nl/avisi/kotlinwebtest/RestDslTest.kt]: true, my bad 👍

**Pull request merged**